tests/Resources/ti.database.test.js

```diff
@@ -621,6 +670,44 @@ describe('Titanium.Database', function () {
 			}
 		});
 
+		it('returns Promise', function (finish) { // eslint-disable-line mocha/no-identical-title
+			this.timeout(5000);
+			const db = Ti.Database.open('execute_all.db');
+
+			const queries = [
+				// Execute a query to create a test table
+				'CREATE TABLE IF NOT EXISTS testTable (text TEXT, number INTEGER)',
+				// Delete any existing data if the table already existed
+				'DELETE FROM testTable',
+				// Insert test data into the table
+				'INSERT INTO testTable (text, number) VALUES (\'John Smith\', 123456789)',
+				// Execute a query to return the rows of the database
+				'SELECT rowid, text, number FROM testTable',
+			];
+
+			const result = db.executeAllAsync(queries);
+			result.should.be.a.Promise();
+			result.then(results => {
+				// the returned results array should be the same length as the input query array
+				should(results.length).eql(queries.length);
+
+				const rows = results[3];
+				try {
+					// Validate the returned 'rows' object
+					should(rows).be.a.Object();
+					should(rows.rowCount).be.eql(1);
+					should(rows.fieldCount).be.eql(3);
+					should(rows.validRow).be.true();
+				} finally {
```

Comment:
<!--
  0 failure: 
  2 warning:  tests/Resources/t..., tests/Resources/t...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/ti.database.test.js;
  Line: 701;
-->

- :warning: tests/Resources/ti.database.test.js line 701 – Each then() should return a value or throw (promise/always-return)
- :warning: tests/Resources/ti.database.test.js line 701 – Each then() should return a value or throw (promise/always-return)